### ISSUE TITLE
[BUG] NgZone reference gets overridden when dependencies are shared

### ISSUE THREAD

**Issue by arturovt** [single-spa/single-spa-angular#206]
After I started investigating _sharing dependencies approach_ in Angular I have noticed that `ngZone` gets overridden by the last mounted application here:

https://github.com/single-spa/single-spa-angular/blob/389a16cb186ed5fd975199ff2e4670084c47a84f/src/src/extra-providers.ts#L73-L75

So if we've got 2 active applications:
* both of them will call `setNgZone(ngZone)`
* since `SingleSpaPlatformLocation` is a part of the `PLATFORM_INJECTOR` then the second application will override `ngZone` with its own instance
* the first application will not be notified about `popstate` events because here:
https://github.com/single-spa/single-spa-angular/blob/389a16cb186ed5fd975199ff2e4670084c47a84f/src/src/extra-providers.ts#L63
`ngZone` is coming from 2nd application.

The solution is to keep references to all `ngZone`'s:

```ts
export class SingleSpaPlatformLocation {
  private ngZones: NgZone[] = [];

  setNgZone(ngZone: NgZone): void {
    this.ngZones.push(ngZone);
  }

  onPopState() {
    // ...
    for (const ngZone of this.ngZones) {
      ngZone.run(() => fn(event));
    }
  }

  destroy(ngZone: NgZone): void {
    // ...
    const ngZoneIndex = this.ngZones.indexOf(ngZone);
    this.ngZones.splice(ngZoneIndex, 1);
  }
}
```

We also have to take into consideration that part of code:

https://github.com/single-spa/single-spa-angular/blob/389a16cb186ed5fd975199ff2e4670084c47a84f/src/src/extra-providers.ts#L33

Since it will clear all `popstate` event listeners even if one of the applications is still active.

**Comment by joeldenning** [comment]
Good catch - yes let's switch to an array of ngZones.

**Comment by filoxo** [comment]
Closing since this was resolved in #262 and released in 4.4.0.
